resolver/dns/dns_resolver.go

```diff
@@ -52,3 +52,9 @@ func SetResolvingTimeout(timeout time.Duration) {
 func NewBuilder() resolver.Builder {
 	return dns.NewBuilder()
 }
+
+// SetMinResolutionRate sets the default minimum rate at which DNS re-resolutions are
+// allowed. This helps to prevent excessive re-resolution.
+func SetMinResolutionRate(d time.Duration) {
```

Comment:
I swear I wrote this comment in the last pass, but it's gone now.  :)  Sorry!

Please rename this to `SetMinResolutionInterval` (and change the global variable name to match).  A "rate" is the opposite of what this is.  (This actually determines the _maximum_ rate, or the _minimum_ interval.)

Reply:
You're totally right, thanks for pointing it out!
I think I managed to change all of them. I triple checked to make sure but let me know if there's still anything wrong.